### PR TITLE
feat: Brevo Email Integration To Share Forms

### PR DESCRIPTION
**Pull request by abanuelo**
### Description

Adding email integration based on design. Below is video illustrating the changes:

https://github.com/cascaritaco/cascarita/assets/32311654/0f04c096-7446-41ed-a1ac-b367ac52c5a0

### Issue Link

[Jira Ticket - C1-103](https://cascarita.atlassian.net/jira/software/c/projects/C1/boards/2/backlog?selectedIssue=C1-103&atlOrigin=eyJpIjoiNDAzOThjZTZmMjFhNGQ2OThmODRlOGVlYjY4YTlmZjMiLCJwIjoiaiJ9)

### Checklist

- [X] I have tested the changes locally by running `npm run test`
- [ ] I have added or updated relevant documentation
- [ ] I have autoformatted the code by running `npm run eslint`
- [ ] I have added test cases (if applicable)

### CODE REVIEW

**Review by elimanzo — CHANGES_REQUESTED** [review]
Hey @abanuelo this is really cool! I have a small nit, but other than that good job!

File: client/pages/Forms/Forms.tsx [hunk 1]
```diff
@@ -98,12 +123,17 @@ const Forms = () => {
...
                 href={form.form_data._links.display}
                 target="_blank"
                 rel="noopener noreferrer">
                 <ShareButton />
-              </a>
+              </a> */}
+              <ShareModal
+                formLink={form.form_data._links.display}
+                open={open}
+                setOpen={setOpen}
```
[the remark this answers is not in the document]
Comment (abanuelo) [reply client/pages/Forms/Forms.tsx:135]: Gotcha! Also look what I saw on my daily.dev newsletter today: https://app.daily.dev/posts/PMBIEyN9J?utm_source=notification&utm_medium=email&utm_campaign=digest
It's like there are signs from the universe or something 🙃

[the remark this answers is not in the document]
Comment (elimanzo) [reply client/pages/Forms/Forms.tsx:135]: 😂 i promise I didn't write that newsletter.

[the remark this answers is not in the document]
Comment (abanuelo) [reply client/pages/Forms/Forms.tsx]: Resolved!

File: client/pages/Forms/Forms.tsx [hunk 2]
```diff
@@ -15,11 +15,48 @@ import {
...
+import ShareForm from "../../components/Forms/ShareForm/ShareForm";
+
+interface ShareModalProps {
+  formLink: string;
+  open: boolean;
+  onOpen: (open: boolean) => void;
+}
+
+const ShareModal: React.FC<ShareModalProps> = ({ formLink, open, onOpen }) => {
+  const handleOpenChange = (isOpen: boolean) => {
```
Comment (elimanzo) [inline client/pages/Forms/Forms.tsx:29]: I think we can get rid of the intermediate variables and just have it as such 

```Typescript
const ShareModal: React.FC<ShareModalProps> = ({ formLink, isOpen, onOpen }) => {
  return (
    <Modal open={isOpen} onOpenChange={onOpen}>
      <Modal.Button asChild className={styles.btn}>
        <button onClick={() => onOpen(true)}>
          <ShareButton />
        </button>
      </Modal.Button>
      <Modal.Content title="Share Form">
        <ShareForm afterClose={() => onOpen(false)} formLink={formLink} />
      </Modal.Content>
    </Modal>
  );
};
```

File: client/pages/Forms/Forms.tsx [hunk 3]
```diff
@@ -15,11 +15,48 @@ import {
...
+import React from "react";
+import ShareForm from "../../components/Forms/ShareForm/ShareForm";
+
+interface ShareModalProps {
+  formLink: string;
+  open: boolean;
+  onOpen: (open: boolean) => void;
+}
+
+const ShareModal: React.FC<ShareModalProps> = ({ formLink, open, onOpen }) => {
```
Comment (elimanzo) [inline client/pages/Forms/Forms.tsx:28]: looking at the variable names, probably best to call `open` to `isOpen`

File: client/pages/Forms/Forms.tsx [hunk 4]
```diff
@@ -15,11 +15,48 @@ import {
...
+      </Modal.Content>
+    </Modal>
+  );
+};
 
 const Forms = () => {
   const { t } = useTranslation("Forms");
   const [sorts, setSorts] = useState("");
   const [forms, setForms] = useState<Form[]>([]);
+  const [open, setOpen] = useState(false);
```
Comment (elimanzo) [inline client/pages/Forms/Forms.tsx:59]: same as above calling `isOpen` and `setIsOpen`

File: client/pages/Forms/Forms.tsx [hunk 5]
```diff
@@ -98,12 +135,11 @@ const Forms = () => {
...
-              <a
-                href={form.form_data._links.display}
-                target="_blank"
-                rel="noopener noreferrer">
-                <ShareButton />
-              </a>
+              <ShareModal
+                formLink={form.form_data._links.display}
+                open={open}
+                onOpen={(open: boolean) => setOpen(open)}
```
Comment (elimanzo) [inline client/pages/Forms/Forms.tsx:141]: same thing as above calling `isOpen`

**Review by elimanzo — CHANGES_REQUESTED** [review]
I just have a couple of nits, and I think we're good to go 😄

File: client/pages/Forms/Forms.tsx (hunk 3 above)
Comment (abanuelo) [reply client/pages/Forms/Forms.tsx]: Ack! Done!

File: client/pages/Forms/Forms.tsx (hunk 4 above)
Comment (abanuelo) [reply client/pages/Forms/Forms.tsx]: Ack! Done!

File: client/pages/Forms/Forms.tsx (hunk 5 above)
Comment (abanuelo) [reply client/pages/Forms/Forms.tsx]: Ack! Done!

File: client/pages/Forms/Forms.tsx (hunk 2 above)
Comment (abanuelo) [reply client/pages/Forms/Forms.tsx]: Ack! Done!

**Review by elimanzo — APPROVED** [review]
LGTM!

**Pull request merged**